ClemBot.Bot/bot/cogs/translate_cog.py

```diff
@@ -146,75 +97,287 @@ async def translate_given_lang(self, ctx, input):
         name = 'Translated to ' + LANGUAGE_SHORT_CODE_TO_NAME[response[0]['translations'][0]['to'].lower()]
         embed.add_field(name=name, value=response[0]['translations'][0]['text'], inline=False)
         await ctx.send(embed=embed)
+        return
 
-    async def translate_detect_lang(self, ctx, input):
-        output_lang = await get_lang_code(self, ctx, input[0])
-        if output_lang == None:
-            return
 
-        text = ' '.join(input[1:])
 
-        output_lang = await get_lang_code(self, ctx, output_lang)
-        log.info(f'Output Lang Code: {output_lang}')
-
-        params = {
-            'api-version': '3.0',
-            'to': output_lang
+def is_valid_lang_code(input: str):
```

Comment:
First, `input` is a built in function in Python, it's generally a bad idea to reassign built in stuff. 

As for the rest of the function, it's really weird. You have dictionaries, why not use them as dictionaries instead of lists?

Perhaps something like this would work?
```py
def is_valid_lang_code(code: str) -> Optional[str]:
    code_lower = code.lower()

    if code_lower in LOWERCASE_LANGUAGENAME:
        return LOWERCASE_LANGUAGENAME[code_lower]

    if code_lower in LOWERCASE_LANGUAGE_SHORT_CODE_TO_NAME:
        return LOWERCASE_LANGUAGE_SHORT_CODE_TO_NAME[code_lower]
```